### DISCUSSION TITLE
Errors when customizing kernel for talos

### DISCUSSION THREAD

**Discussion by yumin-l** [siderolabs/talos#7460] [Q&A] [answered]
I'm customizing kernel for talos with reference to the document(https://www.talos.dev/v1.4/advanced/customizing-the-kernel/). But I got some problems when doing the last step (DOCKER_BUILDKIT=0 docker build --build-arg RM="/lib/modules" -t installer:kernel .). Any ideas for this problem?
![image](https://github.com/siderolabs/talos/assets/101164877/c8bdfb2b-8aee-4d3d-b3fe-b2eb21104100).

**Comment by frezbo** [comment]
seems you're trying from talos 1.5, please follow the Talos 1.5 docs then: https://www.talos.dev/v1.5/advanced/customizing-the-kernel/

**Reply by yumin-l** [reply]
Thanks for your quick reply, it works.
But how can I use the customized installer to build Talos and install it?  I try to use **_talosctl upgrade_** to upgrade my cluster with the customized installer but it didn't work.
![image](https://github.com/siderolabs/talos/assets/101164877/979d2b71-ed57-432a-8e1f-4edd61fb6135)

**Reply by frezbo** [reply, marked as answer]
that seems like an issue with authenticating to your registry, you either need to setup auth for registry in talos or make the image public

**Reply by yumin-l** [reply]
Thanks a lot
